engine/runtime/src/main/java/org/enso/interpreter/caches/Cache.java

```diff
@@ -191,65 +192,64 @@ public final Optional<T> load(EnsoContext context) {
       return spi.getCacheRoots(context)
           .flatMap(
               roots -> {
+                // Load from the global root as a priority.
                 try {
-                  Optional<T> loadedCache;
-                  // Load from the global root as a priority.
-                  loadedCache = loadCacheFrom(roots.globalCacheRoot(), context, logger);
-                  if (loadedCache.isPresent()) {
-                    logger.log(
-                        logLevel,
-                        "Using cache for ["
-                            + logName
-                            + " at location ["
-                            + toMaskedPath(roots.globalCacheRoot()).applyMasking()
-                            + "].");
-                    return loadedCache;
-                  }
-
-                  loadedCache = loadCacheFrom(roots.localCacheRoot(), context, logger);
-                  if (loadedCache.isPresent()) {
+                  var globalCache = loadCacheFrom(roots.globalCacheRoot(), context, logger);
+                  logger.log(
+                      logLevel,
+                      "Using cache for ["
+                          + logName
+                          + " at location ["
+                          + toMaskedPath(roots.globalCacheRoot()).applyMasking()
+                          + "].");
+                  return Optional.of(globalCache);
+                } catch (IOException globalEx) {
+                  try {
+                    var localCache = loadCacheFrom(roots.localCacheRoot(), context, logger);
                     logger.log(
                         logLevel,
                         "Using cache for ["
                             + logName
                             + " at location ["
                             + toMaskedPath(roots.localCacheRoot()).applyMasking()
                             + "].");
-                    return loadedCache;
+                    return Optional.of(localCache);
+                  } catch (IOException localEx) {
+                    logCacheLoadFailure(
```

Comment:
The loading first tries:
- global cache
- if that fails, it remembers the `globalEx` and
- tries local cache
- if that fails, it logs both exceptions

Reply:
I kind of had an impression we wanted the opposite order, i.e. local first and only on failure the global one?
It also sounds a bit more intuitive with other configurations I'm aware of.

Reply:
- reported as #12027
- the bug is still open
- it was reported after the work on this PR was finished
- as such this PR doesn't reflect/fixes the #12027 bug